packages/scoped-custom-element-registry/src/scoped-custom-element-registry.ts

```diff
@@ -11,6 +11,53 @@
  * subject to an additional IP rights grant found at
  * http://polymer.github.io/PATENTS.txt
  */
+
+interface CustomElementConstructor {
+  observedAttributes?: Array<string>;
+  formAssociated?: boolean;
+
+  new (...params: unknown[]): CustomHTMLElement;
+}
+
+interface CustomHTMLElement {
+  connectedCallback?(): void;
+  disconnectedCallback?(): void;
+  adoptedCallback?(): void;
+  attributeChangedCallback?(
+    name: string,
+    oldValue?: string | null,
+    newValue?: string | null,
+    namespace?: string | null
+  ): void;
+  formAssociatedCallback?: unknown;
+  formDisabledCallback?: unknown;
+  formResetCallback?: unknown;
+  formStateRestoreCallback?: unknown;
+}
+
+interface CustomElementRegistry {
```

Comment:
This is an odd name for this interface, and it doesn't match the global type of the same name

Reply:
oh... this is a subtle bit that might require a comment: because this file has no imports or exports, it's global.

Here's TypeScript's (not exactly accurate wrt to EcmaScript) docs:

> In TypeScript, just as in ECMAScript 2015, any file containing a top-level import or export is considered a module. Conversely, a file without any top-level import or export declarations is treated as a script whose contents are available in the global scope (and therefore to modules as well).

So this declaration is _adding_ to the existing global, open `CustomElementRegistry` interface.

This is safe against polluting user global types because we don't publish the .d.ts output.

Reply:
Yeah, this definitely warrants a big comment at the top of the file

Reply:
Added comment.